### PR TITLE
[Bug] Scaleway The volume is created systematically on par1

### PR DESCRIPTION
**Pull request by xilmen**
##### SUMMARY
Change of organization to project, depreciated in the api.
Added missing region parameter.

##### ISSUE TYPE
- Bugfix Pull Request

##### COMPONENT NAME
This module allows to create volumes on the scw datacenter

### CODE REVIEW

File: plugins/modules/cloud/scaleway/scaleway_volume.py [hunk 1]
```diff
@@ -51,10 +51,10 @@
     description:
      - Name used to identify the volume.
     required: true
-  organization:
+  project:
```
[the remark this answers is not in the document]
Comment (remyleone) [reply plugins/modules/cloud/scaleway/scaleway_volume.py:54]: I think you can indeed follow #3951 to add support for a project. I don't think it is necessary to remove organization. Also, I fail to see why you would need a region argument, instance API uses zone such as `fr-par-1` and not region such as `fr-par`. `par1` is deprecated and you should use `fr-par-1` to talk about this zone that belongs to the `fr-par` region.

[the remark this answers is not in the document]
Comment (xilmen) [reply plugins/modules/cloud/scaleway/scaleway_volume.py:54]: I have added the changes to support organization

File: changelogs/fragments/3964-scaleway_volume_add_region.yml [hunk 2]
```diff
@@ -0,0 +1,2 @@
+minor_changes:
+  - scaleway_volume - All volumes are systematically created on par1 (https://github.com/ansible-collections/community.general/pull/3964).
```
Comment (felixfontein) [inline changelogs/fragments/3964-scaleway_volume_add_region.yml:2]:
```suggestion
  - scaleway_volume - all volumes are systematically created on par1 (https://github.com/ansible-collections/community.general/pull/3964).
```
This sounds like you implemented a feature that now all volumes are systematically created on par1. I somehow doubt that's what this PR does?

Also you are not mentioning the new `project` option.

File: plugins/modules/cloud/scaleway/scaleway_volume.py [hunk 3]
```diff
@@ -51,10 +51,14 @@
     description:
      - Name used to identify the volume.
     required: true
+  project:
+    type: str
+    description:
+     - ScaleWay project ID to which volume belongs.
```
Comment (felixfontein) [inline plugins/modules/cloud/scaleway/scaleway_volume.py:57]:
```suggestion
     - ScaleWay project ID to which volume belongs.
   version_added: 4.3.0
```

File: plugins/modules/cloud/scaleway/scaleway_volume.py [hunk 4]
```diff
@@ -51,10 +51,14 @@
...
   organization:
     type: str
     description:
-     - ScaleWay organization ID to which volume belongs.
+     - DEPRECATED ScaleWay organization ID to which volume belongs.
```
Comment (felixfontein) [inline plugins/modules/cloud/scaleway/scaleway_volume.py:61]:
```suggestion
     - ScaleWay organization ID to which volume belongs.
```
We can deprecate it in a follow-up release, but please not in the release where this option is added. Also this is not the way we deprecate options.

File: plugins/modules/cloud/scaleway/scaleway_volume.py [hunk 5]
```diff
@@ -106,11 +110,14 @@
...
 
 def core(module):
+    region = module.params["region"]
     state = module.params['state']
     name = module.params['name']
     organization = module.params['organization']
+    project = module.params['project']
     size = module.params['size']
     volume_type = module.params['volume_type']
+    module.params['api_url'] = SCALEWAY_LOCATION[region]["api_endpoint"]
```
Comment (felixfontein) [inline plugins/modules/cloud/scaleway/scaleway_volume.py:120]: Is this a bugfix? Is it related to the other bugfixes? If not, it should be a separate PR.

File: plugins/modules/cloud/scaleway/scaleway_volume.py [hunk 6]
```diff
@@ -123,14 +130,20 @@ def core(module):
 
     volumeByName = None
     for volume in volumes_json['volumes']:
+        if volume['project'] == project and volume['name'] == name:
+            volumeByName = volume
+
         if volume['organization'] == organization and volume['name'] == name:
```
Comment (felixfontein) [inline plugins/modules/cloud/scaleway/scaleway_volume.py:136]: Does the API return both `organization` and `project` with the same value? Or only one of these two?

File: plugins/modules/cloud/scaleway/scaleway_volume.py (hunk 5 above)
Comment (xilmen) [reply plugins/modules/cloud/scaleway/scaleway_volume.py:121]: I need this part to have the scw aliases. example par1 = fr-par-1

File: plugins/modules/cloud/scaleway/scaleway_volume.py [hunk 7]
```diff
@@ -51,6 +51,11 @@
     description:
      - Name used to identify the volume.
     required: true
+  project:
+    type: str
+    description:
+     - ScaleWay project ID to which volume belongs.
+   version_added: 4.3.0
```
Comment (felixfontein) [inline plugins/modules/cloud/scaleway/scaleway_volume.py:58]:
```suggestion
    version_added: 4.3.0
```

File: plugins/modules/cloud/scaleway/scaleway_volume.py (hunk 6 above)
Comment (xilmen) [reply plugins/modules/cloud/scaleway/scaleway_volume.py:137]: I check with ansible -vvv i have 2 return from API:
organization: null
project: <ID_project>

So I guess if we pass organization has the place of project. The value of project would be null

File: plugins/modules/cloud/scaleway/scaleway_volume.py [hunk 8]
```diff
@@ -51,6 +51,11 @@
     description:
      - Name used to identify the volume.
     required: true
+  project:
+    type: str
+    description:
+     - ScaleWay project ID to which volume belongs.
```
Comment (remyleone) [inline plugins/modules/cloud/scaleway/scaleway_volume.py:57]:
```suggestion
     - Scaleway project ID to which volume belongs.
```

**Comment by remyleone** [comment]
shipit

File: plugins/modules/cloud/scaleway/scaleway_volume.py [hunk 9]
```diff
@@ -123,14 +131,20 @@ def core(module):
...
     for volume in volumes_json['volumes']:
+        if volume['project'] == project and volume['name'] == name:
+            volumeByName = volume
+
         if volume['organization'] == organization and volume['name'] == name:
             volumeByName = volume
 
     if state in ('present',):
+        if project is None:
+            project = organization
```
Comment (felixfontein) [inline plugins/modules/cloud/scaleway/scaleway_volume.py:142]: Shouldn't this assignment happen earlier, and only `project` be used in the above `for` loops?

File: plugins/modules/cloud/scaleway/scaleway_volume.py [hunk 10]
```diff
@@ -134,13 +137,7 @@ def core(module):
         if volume['project'] == project and volume['name'] == name:
```
Comment (felixfontein) [inline plugins/modules/cloud/scaleway/scaleway_volume.py:137]: @remyleone are there cases where `organization` is still returned (and `project` is `null`)?

Comment (remyleone) [reply plugins/modules/cloud/scaleway/scaleway_volume.py:137]: I don't think so.

**Comment by felixfontein** [comment]
In that case, looks good to me as well!

**Comment by felixfontein** [comment]
@xilmen thanks for your contribution!
@remyleone thanks for reviewing!
